src/Pact/Types/Swagger.hs

```diff
@@ -44,33 +48,39 @@ import Data.Text (Text)
 declareGenericEmpty ::
   forall a d f proxy. (Generic a, Rep a ~ D1 d f, Datatype d) =>
   proxy a -> Declare (Definitions Schema) NamedSchema
-declareGenericEmpty = declareGenericSchema mempty
+declareGenericEmpty _ = declareGenericSchema mempty (Proxy @a)
 
 -- | 'ToSchema' generic implementation with provided schema.
 declareGenericSchema ::
   forall a d f proxy. (Generic a, Rep a ~ D1 d f, Datatype d) =>
   Schema -> proxy a -> Declare (Definitions Schema) NamedSchema
-declareGenericSchema s proxy = pure $ genericNameSchema defaultSchemaOptions proxy s
+declareGenericSchema s _ = pure $ genericNameSchema defaultSchemaOptions (Proxy @a) s
 
 declareGenericString ::
   forall a d f proxy. (Generic a, Rep a ~ D1 d f, Datatype d) =>
   proxy a -> Declare (Definitions Schema) NamedSchema
-declareGenericString = declareGenericSchema (schemaOf $ swaggerType SwaggerString)
+declareGenericString _ = declareGenericSchema (schemaOf $ swaggerType SwaggerString) (Proxy @a)
 
-namedSchema :: Text -> Schema -> proxy a -> Declare (Definitions Schema) NamedSchema
+namedSchema ::
+    forall a proxy .
+    Text -> Schema -> proxy a -> Declare (Definitions Schema) NamedSchema
 namedSchema n s _ = return $ NamedSchema (Just n) s
 
 -- | like 'byteSchema' but with (non-standard) "base64url" in format.
 byteBase64url :: Schema
-byteBase64url = set type_ SwaggerString . set format (Just "base64url") $ mempty
+byteBase64url = swaggerType SwaggerString . set format (Just "base64url") $ mempty
 
 fixedLength :: Integral i => i -> Schema -> Schema
 fixedLength i =
   set maxLength (Just $ fromIntegral i) .
   set minLength (Just $ fromIntegral i)
 
 swaggerType :: SwaggerType 'SwaggerKindSchema -> Schema -> Schema
+#if MIN_VERSION_swagger2(2,4,0)
```

Comment:
This is the kind of complexity we want to avoid. Note what you're doing here: you're creating a function just to call a lens that broke between two versions, and using CPP in said function.

The harder problem involves the schema for `ObjectMap` which would require CPP to support `AdditionalProperties` which similarly is completely broken across the upgrade.

Reply:
>  Note what you're doing here: you're creating a function

The code doesn't create a new function. It just makes an existing function work with both versions.

Reply:
I don't consider this overly complex. It's just about injecting a `Just`. This use of C macros is best practice for making Haskell code portable. 

The code compiles with both versions. Are we currently using `AdditionalProperties`? If you can give me a pointer I am happy to take a look.

Reply:
Servant and swagger related packages are notorious for this kind of API breakages. In my experience it's inevitable to maintain some legacy code paths for different versions when using these packages over a longer period of time.

I have had good experience with supporting a small range of recent versions directly at the API usage site (I try to keep it in a single or small number of modules) in order to not let the noise of breaking changes propagate upwards through the code.

Reply:
I think my main reservation here is, it's a change that none of our builds will verify actually works. Until we have a CI for Pact that is importing swagger2 2.4, I don't want to make this change.